src/API/Status/PreviewCard.vala

```diff
@@ -116,11 +121,30 @@ public class Tuba.API.PreviewCard : Entity, Widgetizable {
 	public CardSpecialType card_special_type {
 		get {
 			if (is_peertube) {
-				return CardSpecialType.PEERTUBE;
+				#if CLAPPER
+					try {
+						var uri = GLib.Uri.parse (this.url, GLib.UriFlags.NONE);
+						if (Clapper.enhancer_check (typeof (Clapper.Extractable), "peertube", uri.get_host (), null)) {
```

Comment:
In case of custom schemes ("peertube" here), `host` is not mandatory, thus you should be able to skip URI parsing here to make this somewhat faster (and pass `null` instead of host here).

Reply:
Sounds good! From my testing I'm getting "Extraction failed" on PeerTube videos e.g. https://video.blender.org/w/35c13d53-b767-4e20-a687-6c08192cc7ec (https:// => peertube://); including on Clapper app.

From a quick look, the API response seems fine to me https://video.blender.org/api/v1/videos/35c13d53-b767-4e20-a687-6c08192cc7ec :thinking:

Reply:
Fixed with: https://github.com/Rafostar/clapper-enhancers/commit/be60a2bf304e07134b0080237df49a717846d712